File: next/security/backend-ms1/src/main/java/ma/zs/stocky/ws/converter/security/ModelPermissionUserConverter.java
package  ma.zs.stocky.ws.converter.security;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import ma.zs.stocky.ws.converter.security.UserConverter;
import ma.zs.stocky.ws.converter.security.ModelPermissionConverter;
import ma.zs.stocky.ws.converter.security.ActionPermissionConverter;

import ma.zs.stocky.bean.core.security.User;


import ma.zs.stocky.zynerator.util.StringUtil;
import ma.zs.stocky.zynerator.converter.AbstractConverter;
import ma.zs.stocky.zynerator.util.DateUtil;
import ma.zs.stocky.bean.core.security.ModelPermissionUser;
import ma.zs.stocky.ws.dto.security.ModelPermissionUserDto;

@Component
public class ModelPermissionUserConverter extends AbstractConverter<ModelPermissionUser, ModelPermissionUserDto> {

    @Autowired
    private UserConverter userConverter ;
    @Autowired
    private ModelPermissionConverter modelPermissionConverter ;
    @Autowired
    private ActionPermissionConverter actionPermissionConverter ;
    private boolean actionPermission;
    private boolean modelPermission;
    private boolean user;

    public  ModelPermissionUserConverter() {
        super(ModelPermissionUser.class, ModelPermissionUserDto.class);
    }

    @Override
    public ModelPermissionUser toItem(ModelPermissionUserDto dto) {
        if (dto == null) {
            return null;
        } else {
        ModelPermissionUser item = new ModelPermissionUser();
            if(StringUtil.isNotEmpty(dto.getId()))
                item.setId(dto.getId());
            if(dto.getValue() != null)
                item.setValue(dto.getValue());
            if(StringUtil.isNotEmpty(dto.getSubAttribute()))
                item.setSubAttribute(dto.getSubAttribute());
            if(this.actionPermission && dto.getActionPermission()!=null &&  dto.getActionPermission().getId() != null)
                item.setActionPermission(actionPermissionConverter.toItem(dto.getActionPermission())) ;

            if(this.modelPermission && dto.getModelPermission()!=null &&  dto.getModelPermission().getId() != null)
                item.setModelPermission(modelPermissionConverter.toItem(dto.getModelPermission())) ;

            if(dto.getUser() != null && dto.getUser().getId() != null){
                item.setUser(new User());
                item.getUser().setId(dto.getUser().getId());
                item.getUser().setId(dto.getUser().getId());
            }




        return item;
        }
    }

    @Override
    public ModelPermissionUserDto toDto(ModelPermissionUser item) {
        if (item == null) {
            return null;
        } else {
            ModelPermissionUserDto dto = new ModelPermissionUserDto();
            if(StringUtil.isNotEmpty(item.getId()))
                dto.setId(item.getId());
                dto.setValue(item.getValue());
            if(StringUtil.isNotEmpty(item.getSubAttribute()))
                dto.setSubAttribute(item.getSubAttribute());
            if(this.actionPermission && item.getActionPermission()!=null) {
                dto.setActionPermission(actionPermissionConverter.toDto(item.getActionPermission())) ;
            }
            if(this.modelPermission && item.getModelPermission()!=null) {
                dto.setModelPermission(modelPermissionConverter.toDto(item.getModelPermission())) ;
            }
            if(this.user && item.getUser()!=null) {
                dto.setUser(userConverter.toDto(item.getUser())) ;
            }


        return dto;
        }
    }




    public void initObject(boolean value) {
        this.actionPermission = value;
        this.modelPermission = value;
        this.user = value;
    }


    public UserConverter getUserConverter(){
        return this.userConverter;
    }
    public void setUserConverter(UserConverter userConverter ){
        this.userConverter = userConverter;
    }
    public ModelPermissionConverter getModelPermissionConverter(){
        return this.modelPermissionConverter;
    }
    public void setModelPermissionConverter(ModelPermissionConverter modelPermissionConverter ){
        this.modelPermissionConverter = modelPermissionConverter;
    }
    public ActionPermissionConverter getActionPermissionConverter(){
        return this.actionPermissionConverter;
    }
    public void setActionPermissionConverter(ActionPermissionConverter actionPermissionConverter ){
        this.actionPermissionConverter = actionPermissionConverter;
    }
    public boolean  isActionPermission(){
        return this.actionPermission;
    }
    public void  setActionPermission(boolean actionPermission){
        this.actionPermission = actionPermission;
    }
    public boolean  isModelPermission(){
        return this.modelPermission;
    }
    public void  setModelPermission(boolean modelPermission){
        this.modelPermission = modelPermission;
    }
    public boolean  isUser(){
        return this.user;
    }
    public void  setUser(boolean user){
        this.user = user;
    }
}

File: nest/security-v2/backend-ms1/src/app/shared/converter/security/ModelPermissionUserConverter.ts
import {forwardRef, Inject, Injectable} from "@nestjs/common";
import {AbstractConverter} from "src/app/zynerator/converter/AbstractConverter";

import {ModelPermissionUser} from "src/app/shared/bean/core/security/ModelPermissionUser";
import {ModelPermissionUserDto} from "src/app/shared/dto/security/ModelPermissionUserDto";

import {UserConverter} from "src/app/shared/converter/security/UserConverter";
import {ModelPermissionConverter} from "src/app/shared/converter/security/ModelPermissionConverter";
import {ActionPermissionConverter} from "src/app/shared/converter/security/ActionPermissionConverter";
import {User} from "src/app/shared/bean/core/security/User";

@Injectable()
export class ModelPermissionUserConverter extends AbstractConverter<ModelPermissionUser, ModelPermissionUserDto> {

    actionPermission: boolean;
    modelPermission: boolean;
    user: boolean;

    constructor(
        @Inject(forwardRef(() => UserConverter)) readonly userConverter: UserConverter,
        @Inject(forwardRef(() => ModelPermissionConverter)) readonly modelPermissionConverter: ModelPermissionConverter,
        @Inject(forwardRef(() => ActionPermissionConverter)) readonly actionPermissionConverter: ActionPermissionConverter,
        ) {
            super();
            this.init(true);
    }

    toItem(dto: ModelPermissionUserDto): ModelPermissionUser {
        if (!dto) {
            return null;
        }
        const item =new ModelPermissionUser();
        if (dto.id) {
            item.id = dto.id;
        }
        if (dto.value) {
            item.value = dto.value;
        }
        if (dto.subAttribute) {
            item.subAttribute = dto.subAttribute;
        }


		if(this.actionPermission && dto.actionPermission &&  dto.actionPermission.id)
			item.actionPermission = this.actionPermissionConverter.toItem(dto.actionPermission) ;

		if(this.modelPermission && dto.modelPermission &&  dto.modelPermission.id)
			item.modelPermission = this.modelPermissionConverter.toItem(dto.modelPermission) ;

		if(dto.user && dto.user.id){
			item.user = new User();
			item.user.id = dto.user.id;
			item.user.id =  dto.user.id;
		}



        return item;
    }

    toDto(item: ModelPermissionUser): ModelPermissionUserDto {
        if (!item) {
            return null;
        }
        const dto = new ModelPermissionUserDto();

        if (item.id) {
            dto.id = item.id;
        }
        if (item.value) {
            dto.value = item.value;
        }
        if (item.subAttribute) {
            dto.subAttribute = item.subAttribute;
        }
        if(this.actionPermission && item.actionPermission)
            dto.actionPermission = this.actionPermissionConverter.toDto(item.actionPermission) ;
        if(this.modelPermission && item.modelPermission)
            dto.modelPermission = this.modelPermissionConverter.toDto(item.modelPermission) ;
        if(this.user && item.user)
            dto.user = this.userConverter.toDto(item.user) ;


        return dto;
    }



    public initObject(value: boolean): void {
        this.actionPermission = value;
        this.modelPermission = value;
        this.user = value;
    }
}
